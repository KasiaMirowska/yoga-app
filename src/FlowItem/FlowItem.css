.flow-item {
    background-color: white; 
    margin: 25px;
    text-align: center;
    padding: 20px;
    border-radius: 10px;
    filter: drop-shadow(0 0 0.75rem rgb(151, 132, 136));
    width: 15%;
    max-height: 60px; 
}

.li{
    height: 100%;
    display: flex;
    flex-direction: column;
}

.img-container{
    flex: 1 1 auto;
    max-height: 80px;      
}

.flow-img {
   max-height: 60px; 
   padding: 0;
   margin: auto
}

footer {
    text-align: right; 
    width: 100%;
}

.flow-item-button {
    color: rgb(91, 128, 149);
    background-color: white;
    padding: 0;
}

@media (min-width: 800px) {
    .flow-item {
        width: 10%;
        max-height: 140px; 
        border-radius: 10px;
    }

    .flow-img {
        max-height: 120px;    
    }

    .img-container{
        flex: 1 1 auto;
        max-height: 125px;    
    }
}  