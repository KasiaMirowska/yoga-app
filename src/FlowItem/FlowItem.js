import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import YogaContext from '../Context';

export default class FlowItem extends React.Component {
    static contextType = YogaContext;

    linkTo = () => {
        const { flowId, poseId } = this.props;
        this.props.history.push(`/flow/${flowId}/pose/${poseId}`)
    }

    deletePoseFromFlow = (id) => {
        console.log('deleting', id)
        this.context.deletePoseFromFlow(id)
    }
    
    render() {
        return (
            <li key={this.props.id}>
                <img src={this.props.img} onClick={this.linkTo}/>
                <button type='button' onClick={() => this.deletePoseFromFlow(this.props.poseId)}>
                    <FontAwesomeIcon icon={faTrashAlt} />
                </button>
            </li>
        )
    }

}
;