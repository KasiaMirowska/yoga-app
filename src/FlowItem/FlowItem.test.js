import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import FlowItem from './FlowItem';
import {BrowserRouter as BR} from 'react-router-dom';

describe('FlowItem component', () => {
    let props = {
        flowId: 1,
        poseId: 1,
        history: {location: {pathname: `/flow/1/pose/1}`}},
        
    }
   
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><FlowItem {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><FlowItem {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })

})