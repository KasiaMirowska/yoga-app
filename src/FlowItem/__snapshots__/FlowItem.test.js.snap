// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowItem component renders UI as expected 1`] = `
<div
  className="flow-item"
>
  <li
    className="li"
  >
    <div
      className="img-container"
    >
      <img
        alt="yoga asana"
        className="flow-img"
        onClick={[Function]}
      />
    </div>
    <footer>
      <button
        className="flow-item-button"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 "
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </footer>
  </li>
</div>
`;
